java/ddlogapi.c

```diff
@@ -433,6 +433,14 @@ JNIEXPORT jint JNICALL Java_ddlogapi_DDlogAPI_ddlog_1get_1table_1id(
     return (jint)id;
 }
 
+JNIEXPORT jstring JNICALL Java_ddlogapi_DDlogAPI_ddlog_1get_1table_1name(
+    JNIEnv *env, jclass class, jint id) {
+    const char* table = ddlog_get_table_name(id);
```

Comment:
What happens if `table==NULL`? Will it return a Java `null`? Shouldn't it throw an exception in this case?

Reply:
There are two cases it could return null: one is that unwrap failed, which I am not sure what it means, and when the id is wrong (and ddlogt_get_table_name returns null), which would warrant an exception. I don't know how you can tell them apart, though. Ideally the unwrap_or_else would set up the error message, but I don't have a handle to the HDDlog in that function.

Reply:
I think unwrap will fail if the string cannot be converted to a C string, which should only be possible it its utf8 encoding contains `0` characters. Not sure if it's possible in practice, but in any case I think it's better to return a generic error message like "ddlog_get_table_name failed" than to crash the program.

Reply:
how do I return this error message from rust without a handle to HDDlog?

Reply:
You can return it directly from `ddlogapi.c` by passing it as the second argument to `throwDDlogException`.

Reply:
I can do that, but it would be nicer to distinguish between these two cases. The message should really be set in Rust, not in C, where it's too late.

Reply:
That's true. Ok, let me see if I can fix it. I'll push a commit to your branch...

Reply:
This is not a high value thing. If you spend more than 5 minutes on this let's just forget it; we can file an issue if you really want to do this at some point.